### PR TITLE
Content dependencies

### PR DESCRIPTION
**Pull request by Anclagen**
Parsed structure now outputs roughly like so;
```js
[
 {
  title: "name of institute",
  type: "institute/programme/..."
  foldername: "folder its in for current nav update",
  browserPath: "content/foldername",
  dependency: "title of another course/module/lesson",
  children: [ {}, {} ]  // array of children following this format with their own children
 }, 
 { .... }
]
```
The child arrays are sorted as they are mapped in the parsing structure with the expectation that the first has a null dependency and the others dependent are linked in a chain, the fallback is to just append anything without a dependency at the end. 

All navs have been updated to accept this new data, and will also underline the current page on the nav.

### Summary by OpenAI

**Release Notes**

- Refactor: Navigation components now accept a new data structure and underline the current page on the nav.
- Feature: `parseMarkdown` function now returns an array of objects with arrays of children, sorted by dependency.
- Feature: Added content for a new online learning institution called Hackademics.
- Update: Metadata added to a course, including its type and tags, and changing the type of the institute from "Institute" to "institution".
- Update: `getCurrent` function now accepts an array of `EntityMeta` objects instead of a generic object.
- Feature: Addition of new courses and modules with metadata, tags, and summaries.
- Refactor: Dynamic properties added to the `Entity` class.

> A new institution has arrived,
> With courses and modules to thrive.
> Navigation's been refactored,
> And sorting's been reordered.
> With metadata and summaries galore,
> Learning's never been easier before.

### CODE REVIEW

File: packages/lms/src/lib/components/navigation/CourseNav.svelte [hunk 1]
```diff
@@ -31,22 +36,27 @@
 		font-weight: 600rem;
```
Comment (pretzL) [inline packages/lms/src/lib/components/navigation/CourseNav.svelte:36]: `font-weight: 600rem`
Seems a bit excessive, no?

**Review by pretzL — APPROVED** [review]

File: packages/lms/src/lib/components/navigation/CourseNav.svelte (hunk 1 above)
Comment (Fermain) [reply packages/lms/src/lib/components/navigation/CourseNav.svelte:36]: MORE COWBELL

Comment (Anclagen) [reply packages/lms/src/lib/components/navigation/CourseNav.svelte:36]: cowbell

**Review by Jenfer93 — APPROVED** [review]
Nice work!

File: packages/lms/src/routes/+layout.svelte [hunk 2]
```diff
@@ -17,18 +17,17 @@
...
 		<slot />
 	</Main>
-	<Navigation />
 	<ContentNav />
 	<Footer>
-		<Ego/>
+		<Ego />
 		<Reader />
 	</Footer>
 </div>
```
Comment (Melisa-Zorraindo) [inline packages/lms/src/routes/+layout.svelte:33]: `import Navigation from '$lib/components/navigation/Navigation.svelte';` no longer needed, and it breaks the UI (at least in my local environment)

File: packages/types/lib/Entity.ts [hunk 3]
```diff
@@ -56,17 +59,19 @@ export class Entity implements EntityMeta {
...
     this.slug = path.dirname(address);
 
     // Retrieve entity metadata from disk
-    const { type, title, tags, previous } = getEntityMetaSync(address);
+    const { type, title, tags, previous, ...dynamic } = getEntityMetaSync(address);
     this.type = type;
     this.title = title;
     this.tags = tags;
     this.previous = previous;
+    Object.assign(this, dynamic);
```
[the remark this answers is not in the document]
Comment (Anclagen) [reply packages/types/lib/Entity.ts:75]: Probably, but later.

File: packages/lms/src/lib/components/navigation/ContentNav.svelte [hunk 4]
```diff
@@ -39,45 +29,36 @@
...
-			isCourse = false;
-		}
-		updatePath();
-	});
+
+		page.subscribe((data) => {
+			updatePath();
+		});
+		isCourse = current?.type === 'course';
+	}
```
[the remark this answers is not in the document]
Comment (Anclagen) [reply packages/lms/src/lib/components/navigation/ContentNav.svelte:46]: ??

File: packages/lms/src/lib/components/navigation/ContentNav.svelte [hunk 5]
```diff
@@ -39,45 +29,36 @@
...
-		});
-
 		updatePath();
+		current = getCurrent(institutes, pathArray);
 	}
 
-	$: page.subscribe((data) => {
+	$: {
+		institutes = $files;
 		current = getCurrent(institutes, pathArray);
```
[the remark this answers is not in the document]
Comment (Anclagen) [reply packages/lms/src/lib/components/navigation/ContentNav.svelte]: its working I ain't fiddling with it further.

File: packages/lms/src/lib/components/navigation/ContentNav.svelte [hunk 6]
```diff
@@ -39,45 +29,36 @@
...
-			const unsubscribe = files.subscribe((data) => {
-				institutes = data;
-			});
-
-			return () => unsubscribe();
-		});
-
 		updatePath();
+		current = getCurrent(institutes, pathArray);
 	}
```
[the remark this answers is not in the document]
Comment (Anclagen) [reply packages/lms/src/lib/components/navigation/ContentNav.svelte:35]: eh?

File: packages/lms/src/lib/components/navigation/ContentNav.svelte [hunk 7]
```diff
@@ -7,15 +7,11 @@
...
-		frontmatter?: Frontmatter;
-		// Add other properties as needed, e.g., content, slug, etc.
-	}
+	let institutes: [] | null = [];
+	let current: ContentObject = {};
+	let isCourse = false;
+	let pathArray: string[];
+	let currentPath: string;
 
 	onMount(async () => {
```
[the remark this answers is not in the document]
Comment (Anclagen) [reply packages/lms/src/lib/components/navigation/ContentNav.svelte:17]: Maybe I want to check for null?

File: packages/lms/src/lib/components/navigation/ContentNav.svelte [hunk 8]
```diff
@@ -7,15 +7,11 @@
...
-	}
-
-	interface ContentObject {
-		frontmatter?: Frontmatter;
-		// Add other properties as needed, e.g., content, slug, etc.
-	}
+	let institutes: [] | null = [];
+	let current: ContentObject = {};
+	let isCourse = false;
+	let pathArray: string[];
```
[the remark this answers is not in the document]
Comment (Anclagen) [reply packages/lms/src/lib/components/navigation/ContentNav.svelte]: no

File: packages/lms/src/lib/components/navigation/CourseNav.svelte (hunk 1 above)
Comment (Anclagen) [reply packages/lms/src/lib/components/navigation/CourseNav.svelte]: knocked it down to 500

File: packages/lms/src/lib/components/navigation/CourseNav.svelte [hunk 9]
```diff
@@ -1,27 +1,54 @@
...
+						style="padding-left: {indent + 2}rem"
+						class={currentPath === lesson.foldername ? 'current' : ''}>{lesson.title}</a
 					>
-				{:else}
-					<a href={`${path}${lesson.path}`} style="padding-left: {indent + 1}rem">Overview</a>
-				{/if}
-			{/each}
+				{/each}
+			</div>
 		{/if}
```
[the remark this answers is not in the document]
Comment (Anclagen) [reply packages/lms/src/lib/components/navigation/CourseNav.svelte:52]: Just no!

File: packages/lms/src/routes/content/The Burnt Toast Institute/Culinary Basics/Cookery 101/The Basics/Gastronomy/+page.md [hunk 10]
```diff
@@ -1,9 +1,8 @@
 ---
 title: Gastronomy
 slug: gastronomy
-type: Lesson
-path: The Burnt Toast Institute/Culinary Basics/Cookery 101/The Basics/Gastronomy
-dependency: Baking
+type: lesson
```
[the remark this answers is not in the document]
Comment (Anclagen) [reply packages/lms/src/routes/content/The Burnt Toast Institute/Culinary Basics/Cookery 101/The Basics/Gastronomy/+page.md:4]: oh lord help me

File: packages/lms/src/lib/components/navigation/CourseNav.svelte [hunk 11]
```diff
@@ -1,52 +1,84 @@
...
 	{/each}
```
[the remark this answers is not in the document]
Comment (Anclagen) [reply packages/lms/src/lib/components/navigation/CourseNav.svelte:53]: Thanks I guess

File: packages/lms/src/routes/content/The Burnt Toast Institute/Culinary Basics/Cookery 101/The Basics/Gastronomy/+page.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (Fermain) [reply packages/lms/src/routes/content/The Burnt Toast Institute/Culinary Basics/Cookery 101/The Basics/Gastronomy/+page.md:4]: The red zone is for loading and unloading only

**Review by Fermain — APPROVED** [review]

**Pull request merged**